quinn-proto/src/connection/stats.rs

```diff
@@ -17,6 +17,25 @@ impl std::fmt::Debug for PacketTransmissionStats {
     }
 }
 
+/// Statistics about packets received on a connection
+#[derive(Default)]
+#[non_exhaustive]
+pub struct PacketReceptionStats {
+    /// The amount of packets transmitted on a connection
+    pub packets: u64,
+    /// The total amount of bytes transmitted on a connection
+    pub bytes: u64,
+}
+
+impl std::fmt::Debug for PacketReceptionStats {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("PacketReceptionStats")
+            .field("packets", &self.packets)
+            .field("bytes", &self.bytes)
+            .finish()
+    }
+}
```

Comment:
Does this differ from what you'd get from `derive`?

Reply:
It doesn't at at the moment. Guess I'm still looking for my calculated metrics 😥

Reply:
I think using a derived `Debug` impl and a separate, handwritten, pretty-print-with-derived-metrics function/method could be a fine approach. Deriving `Debug` is so easy that it's not really any extra complexity, and it makes it more natural to prototype or replace downstream.